app/controllers/api/v4/drug_stocks_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Api::V4::DrugStocksController < APIController
+  def index
+    if for_end_of_month.nil?
+      render status: 400, json: {errors: ["valid `date` must be provided - eg. 2021-10-29"]}
+      return
+    end
+
+    @drug_stocks = DrugStock.latest_for_facilities([current_facility], for_end_of_month)
+
+    head :not_found if @drug_stocks.empty?
+  end
+
+  private
+
+  def for_end_of_month
+    @for_end_of_month ||= Date.parse(params[:date]).end_of_month
+  rescue Date::Error, TypeError
```

Comment:
Is the `TypeError` to rescue non-string params being passed? Should we coerce `params[:date]` to a string first in that case?

Reply:
The TypeError is to catch `nil` - ie, date was not provided.